src/manage_sql_secinfo.c

```diff
@@ -2786,35 +2787,35 @@ insert_cve_from_entry (element_t entry, element_t last_modified,
 
   if (base_metrics == NULL)
     {
-      score = NULL;
+      score = 0;
       cvss_vector = NULL;
     }
   else
     {
-      score = element_child (base_metrics,
-                             cvss_is_v3 ? "cvss3:base-score" : "cvss:score");
-      if (score == NULL)
-        {
-          g_warning ("%s: cvss:score missing", __func__);
-          g_free (id);
-          return -1;
-        }
+      double base;
 
       cvss_vector = element_child (base_metrics,
                                    cvss_is_v3 ? "cvss3:vector-string"
                                               : "cvss:vector-string");
+      if (cvss_is_v3 && (cvss_vector == NULL))
+        /* At least nvdcve-2.0-2004.xml uses this format. */
+        cvss_vector = element_child (base_metrics, "cvss3:vectorString");
       if (cvss_vector == NULL)
         {
-          g_warning ("%s: cvss:access-vector missing", __func__);
+          g_warning ("%s: cvss:vector-string missing (id %s)", __func__, id);
           g_free (id);
           return -1;
         }
-    }
 
-  if (score == NULL)
-    score_int = 0;
-  else
-    score_int = round (atof (element_text (score)) * 10);
+      base = get_cvss_score_from_base_metrics (element_text (cvss_vector));
```

Comment:
I'm not sure if we should always calculate the score from the metrics or just use that as a fallback.